BranchMonsterFactory/AppDelegate.m

```diff
@@ -8,94 +8,128 @@
+    
+    [branch initSessionWithLaunchOptions:launchOptions andRegisterDeepLinkHandlerUsingBranchUniversalObject:^(BranchUniversalObject *BUO, BranchLinkProperties *linkProperties, NSError * error) {
         
-        // If the key 'monster' is present in the deep link dictionary
-        // then load the monster viewer with the appropriate monster parameters
-        if ([params objectForKey:@"monster"]) {
-            [MonsterPreferences setMonsterName:[params objectForKey:@"monster_name"]];
-            [MonsterPreferences setFaceIndex:[[params objectForKey:@"face_index"] intValue]];
-            [MonsterPreferences setBodyIndex:[[params objectForKey:@"body_index"] intValue]];
-            [MonsterPreferences setColorIndex:[[params objectForKey:@"color_index"] intValue]];
+        if (BUO && [BUO.metadata objectForKey:@"monster"]) {
             
-            // Choose the monster viewer as the next view controller
-            nextVC = [storyboard instantiateViewControllerWithIdentifier:@"MonsterViewerViewController"];
+            BranchUniversalObject* receivedMonster = BUO;
             
+            NSLog(@"\n\nJust retrieved data from server: %@\n\n", receivedMonster);
             
-        // Else, the app is being opened up from the home screen or from the app store
-        // Load the next logical view controller
-        } else {
-            
-            // If a name has been saved in preferences, then this user has already created a monster
-            // Load the viewer
-            if (![MonsterPreferences getMonsterName]) {
-                [MonsterPreferences setMonsterName:@""];
+            if (self.justLaunched) {
                 
-                // Choose the monster viewer as the next view controller
-                nextVC = [storyboard instantiateViewControllerWithIdentifier:@"MonsterCreatorViewController"];
+                self.justLaunched = NO; //turn off
                 
-            // If no name has been saved, this user is new, so load the monster maker screen
-            } else {
-                nextVC = [storyboard instantiateViewControllerWithIdentifier:@"MonsterViewerViewController"];
+                if (receivedMonster) {
+                    self.initialMonster = receivedMonster;
+                    [[NSNotificationCenter defaultCenter] postNotificationName:@"pushEditAndViewerViews" object:nil];
+                } else {
+                    self.initialMonster = [self emptyMonster];
+                    [[NSNotificationCenter defaultCenter] postNotificationName:@"pushEditView" object:nil];
+                }
+            } else if(self.foregrounded) {
+                if (receivedMonster) {
+                    self.initialMonster = receivedMonster;
+                    self.foregrounded = NO;
+                    [[NSNotificationCenter defaultCenter] postNotificationName:@"pushEditAndViewerViews" object:nil];
+                }
             }
+            
+        } else {
```

Comment:
the effect of this `else` block is to push new view controllers, wiping out the monster i was working on, even if i just backgrounded the app for a second. I think we should re-think this